src/main/java/pro/trevor/tankgame/state/board/unit/GenericTank.java

```diff
@@ -11,16 +11,20 @@
 
 public class GenericTank extends GenericElement implements IMovable, ITickElement, IPlayerElement, IUnit {
 
+    private static final String typeValue = "tank";
     private final String player;
+    private Position position;
 
     public GenericTank(String player, Position position, Map<String, Object> defaults) {
-        super(position, defaults);
+        super(defaults);
         this.player = player;
+        this.position = position;
     }
 
     public GenericTank(JSONObject json) {
         super(json);
         this.player = json.getString("name");
+        this.position = new Position(json.getString("position"));
```

Comment:
This code gets duplicated a lot.  Having a GenericPositionedElement might be a good idea.

Reply:
I've already pointed out that `IElement`, `IUnit`, `IFloor`, `IPositioned`, etc. need to be clarified. To my understanding, `IElement`s should have a reportable position, since they are "board elements". 

Additionally, I think it's detail-leaking for all of the State types/classes to know that they are reported via JSON. A translation layer that the State component doesn't depend on could be better.

Basically, I would like to defer this until we have a design in place.